src/spring-cloud/azext_spring_cloud/app_managed_identity.py

```diff
@@ -0,0 +1,392 @@
+from time import sleep
+
+
+logger = get_logger(__name__)
+
+
+ENABLE_LOWER = "enable"
+DISABLE_LOWER = "disable"
+UPDATING_LOWER = "updating"
+DELETING_LOWER = "deleting"
+APP_CREATE_OR_UPDATE_SLEEP_INTERVAL = 2
+
+
+def app_identity_assign(cmd,
+                        client,
+                        resource_group,
+                        service,
+                        name,
+                        role=None,
+                        scope=None,
+                        system_assigned=None,
+                        user_assigned=None):
+    """
+    Note: Always use sync method to operate managed identity to avoid data inconsistency.
+    :param role: role name of role assignment for system-assigned managed identity.
+    :param scope: scope of role assignment for system-assigned managed identity.
+    :param system_assigned: 1. None or False: Don't change system-assigned managed identity.
+                            2. Enable system-assigned managed identity on app.
+    :param user_assigned: 1. None: Don't change user-assigned managed identities.
+                          2. A non-empty list of user-assigned managed identity resource id to app.
+                          3. A empty list: should be blocked by validator.
+    """
+    # TODO(jiec): Retire legacy identity assign after migration.
+    poller = None
+    if _is_legacy_identity_assign(system_assigned, user_assigned):
+        poller = _legacy_app_identity_assign(cmd, client, resource_group, service, name)
+    else:
+        poller = _new_app_identity_assign(cmd, client, resource_group, service, name, system_assigned, user_assigned)
+    wait_till_end(poller)
+    poller.result()
+    if "succeeded" != poller.status().lower():
+        return poller
+
+    if role and scope:
+        _create_role_assignment(cmd, client, resource_group, service, name, role, scope)
+
+    return client.apps.get(resource_group, service, name)
+
+
+def app_identity_remove(cmd,
+                        client,
+                        resource_group,
+                        service,
+                        name,
+                        system_assigned=None,
+                        user_assigned=None):
+    """
+    Note: Always use sync method to operate managed identity to avoid data inconsistency.
+    :param system_assigned: 1) None or False: Don't change system-assigned managed identity.
+                            2) True: remove system-assigned managed identity
+    :param user_assigned: 1) None: Don't change user-assigned managed identities.
+                          2) An empty list: remove all user-assigned managed identities.
+                          3) A non-empty list of user-assigned managed identity resource id to remove.
+    """
+    app = client.apps.get(resource_group, service, name)
+    if _app_not_updatable(app):
+        raise CLIError("Failed to remove managed identities since app is in {} state.".format(app.properties.provisioning_state))
```

Comment:
Could we use the specific error type instead of `CLIError`?

Reply:
Sure